module/counters/persistent_strict_counters/persistent_strict_monotonic_counter.go

```diff
@@ -0,0 +1,67 @@
+package persistent_strict_counters
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/dgraph-io/badger/v2"
+
+	"github.com/onflow/flow-go/module/counters"
+	"github.com/onflow/flow-go/storage"
+	bstorage "github.com/onflow/flow-go/storage/badger"
+)
+
+// PersistentStrictMonotonicCounter represents the consumer progress with strict monotonic counter.
+type PersistentStrictMonotonicCounter struct {
+	consumerProgress *bstorage.ConsumerProgress
```

Comment:
Rather than moving this to a separate directory, how about creating an interface for `ConsumerProgress`, and defining it in the base `module` directory. Then you can pass the consumer progress into `NewPersistentStrictMonotonicCounter`

Reply:
We already have interface for `ConsumerProgress` and it is based in [`storage`](https://github.com/onflow/flow-go/blob/master/storage/consumer_progress.go#L4) module, so I used that one.

Reply:
[79bd069](https://github.com/onflow/flow-go/pull/5752/commits/79bd069d11de7c6e7395405583380b0648397b7d)